RELEASE.md

```diff
@@ -1,66 +1,79 @@
-the job must be able to find a new tag (one that hasnt already been released), matching the crate name of the role to be
-released. For example, imagine there is currently a release for pool_sv2-v1.0.0. For the jobs releasing a new pool version to succeed,
-it must be able to find a tag containing the string "pool_sv2" with a version greater than 1.0.0. This should not be an issue since the tags
-are automatically created by the `autoversion.yaml` workflow, and moved into main when the generated PR is merged.
+Although our strategy is very similar to the classic gitflow model, we do not keep release branches.
 
-### Github Release Issue
+![](git-branching.png)
 
-Currently we do not support windows releases because we are unable to run the step to return the last tag for a given crate name. See the section above
-for an explaination of the process.
+## Principal Branches
 
+The SRI repo holds two principal branches with an infinite lifetime:
+- `main`
+- `dev`
 
+We consider `main` to be the branch where the source code of `HEAD` always reflects a production-ready state.
 
-## Github Binary Releases
+We consider `dev` to be the branch where the source code of `HEAD` always reflects a state with the latest delivered development changes for the next release.
 
-# Versioning
+Every 4 weeks all the changes to `dev` are merged back into `main` and then tagged with a release number while bumping `MAJOR` and/or `MINOR`.
+
+## Feature Branches
+
+New features are developed into separate branches that only live in the contributor's forks.
+
+- branch off from: `dev`
+- merge back into: `dev`
+- naming convention: `feat-x`, where `x` describes the feature
+
+## Patch Branches
+
+Bugs are patched into separate branches that only live in the contributor's forks.
+
+- branch off from: `main`
+- merge back into: `main` + `dev`
+- naming convention: `patch-x`, where `x` describes the bug/patch
+
+# Releasing Roles Binaries
+
+The [release page of SRI repo](https://github.com/stratum-mining/stratum/releases) provides executable binaries for all SRI roles, targeting popular hardware architectures.
+
+The GitHub binary releases of the roles are handled in the `release-bin.yaml` workflow.
+
+This workflow is manually started by navigating to the "Actions" tab in the SRI repo, then navigating to the Release workflow section, and clicking "Run Workflow".
+
+Note: in order to be able to manually trigger the "Run Workflow" button, the user needs to have "Write" permissions on the repository, otherwise the button will not show up on the UI.
+
+# Publishing Library Crates
+
+Although SRI has a global release cycle, which is attached to the binaries, each internal crate also has its own versioning history.
+
+Lib crates are published to crates.io in the `release-lib.yaml` workflow. The workflow tries to update all the library crates. 
+If a crate is not to updated, the step will fail for that each step have continue-on-error set to true.
```

Comment:
If a create is not to BE updated (I guess it's a typo)

Reply:
tbh the whole phrase was hard to make sense of, so I just re-wrote it on https://github.com/stratum-mining/stratum/pull/766/commits/985c438e5135334cb9449af9b2db00918fa4fbb3

Reply:
> tbh the whole phrase was hard to make sense of, so I just re-wrote it on [985c438](https://github.com/stratum-mining/stratum/commit/985c438e5135334cb9449af9b2db00918fa4fbb3)

Just saw it, now it's perfect 👍